src/protocols/secure_channel/tests/TestDefaultSessionResumptionStorage.cpp

```diff
@@ -109,6 +189,69 @@ void TestSave(nlTestSuite * inSuite, void * inContext)
+    // many times.
+    for (auto & vector : vectors)
+    {
+        NL_TEST_ASSERT(inSuite,
+                       sessionStorage.Save(vector.node, vector.resumptionId, vector.sharedSecret, vector.cats) == CHIP_NO_ERROR);
+    }
+
+    // Read back and verify that only the last record for each node was retained.
+    for (size_t i = ARRAY_SIZE(vectors) - ARRAY_SIZE(nodes); i < ARRAY_SIZE(vectors); ++i)
+    {
+        chip::ScopedNodeId outNode;
+        chip::SessionResumptionStorage::ResumptionIdStorage outResumptionId;
+        chip::Crypto::P256ECDHDerivedSecret outSharedSecret;
+        chip::CATValues outCats;
+
+        // Verify retrieval by node.
+        NL_TEST_ASSERT(inSuite,
+                       sessionStorage.FindByScopedNodeId(vectors[i].node, outResumptionId, outSharedSecret, outCats) ==
+                           CHIP_NO_ERROR);
+        NL_TEST_ASSERT(inSuite,
+                       memcmp(vectors[i].resumptionId.data(), outResumptionId.data(), vectors[i].resumptionId.size()) == 0);
+        NL_TEST_ASSERT(inSuite, memcmp(vectors[i].sharedSecret.Bytes(), outSharedSecret, vectors[i].sharedSecret.Length()) == 0);
+        NL_TEST_ASSERT(inSuite, vectors[i].cats.values[0] == outCats.values[0]);
+        NL_TEST_ASSERT(inSuite, vectors[i].cats.values[1] == outCats.values[1]);
+        NL_TEST_ASSERT(inSuite, vectors[i].cats.values[2] == outCats.values[2]);
+
+        // Validate retreiveal by resumption ID.
+        NL_TEST_ASSERT(inSuite,
+                       sessionStorage.FindByResumptionId(vectors[i].resumptionId, outNode, outSharedSecret, outCats) ==
+                           CHIP_NO_ERROR);
+        NL_TEST_ASSERT(inSuite, vectors[i].node == outNode);
+        NL_TEST_ASSERT(inSuite, memcmp(vectors[i].sharedSecret.Bytes(), outSharedSecret, vectors[i].sharedSecret.Length()) == 0);
+        NL_TEST_ASSERT(inSuite, vectors[i].cats.values[0] == outCats.values[0]);
+        NL_TEST_ASSERT(inSuite, vectors[i].cats.values[1] == outCats.values[1]);
+        NL_TEST_ASSERT(inSuite, vectors[i].cats.values[2] == outCats.values[2]);
+    }
+
+    // Remove all records for all fabrics.  If all three tables of (index, state,
+    // links) are in sync, deleting for each fabric should clean error free.
+    for (const auto & node : nodes)
+    {
+        NL_TEST_ASSERT(inSuite, sessionStorage.DeleteAll(node.GetFabricIndex()) == CHIP_NO_ERROR);
+    }
+
+    // Verify that no entries can be located any longer for any node or
+    // resumption ID.
+    for (auto & vector : vectors)
+    {
+        chip::ScopedNodeId outNode;
+        chip::SessionResumptionStorage::ResumptionIdStorage outResumptionId;
+        chip::Crypto::P256ECDHDerivedSecret outSharedSecret;
+        chip::CATValues outCats;
+
+        // Verify all records for all nodes are gone.
+        NL_TEST_ASSERT(inSuite,
+                       sessionStorage.FindByScopedNodeId(vector.node, outResumptionId, outSharedSecret, outCats) != CHIP_NO_ERROR);
```

Comment:
WIth the TestPersistentStorageDelegate, we also have `GetNumKeys()` and `DumpKeys()` which can be used to do "before"/"after" delta of numbers of key stored, or dump to look with your eyes. Not saying you need to add this here, but we use it in FabricTable tests to make sure our storage backends don't leak stuff after removals.

Reply:
Yes, I am adding code to check for this right now actually.  I'm just using the approach of using the key literals since these are well known in this case.  But that should be fine here.

Will be up shortly.

Reply:
Done.